comp/networkpath/npcollector/npcollectorimpl/npcollector.go

```diff
@@ -337,6 +342,16 @@ func (s *npCollectorImpl) runTracerouteForPath(ptest *pathteststore.PathtestCont
 	path.Namespace = s.networkDevicesNamespace
 	path.Origin = payload.PathOriginNetworkTraffic
 
+	if s.blacklistScanner.ShouldBlacklist(&path) {
```

Comment:
I think "runTracerouteForPath" should not be responsible for this new logic.

I would suggestion replace `s.runTracerouteForPath` with `s.processPathtest()`, then in `processPathtest()` have:

- a/ `s.runTracerouteForPath()` that only runs the traceroute and returns the result pathtrace without sending the data to event platform
- b/ evaluate if the result pathtrace should be discarded
- c/ `s.forwardToEventPlatform()`: send payload to event platform